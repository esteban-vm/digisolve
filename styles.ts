import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
