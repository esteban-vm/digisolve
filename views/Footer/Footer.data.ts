import type { NavLinkProps } from './NavLink'
import type { SocialLinkProps } from './SocialLink'

export const navLinks: NavLinkProps[] = [
  {
    id: '97e51480-e825-11ed-b24d-893619994521',
    text: 'About us',
  },
  {
    id: '9c0412a0-e825-11ed-b24d-893619994521',
    text: 'Blog',
  },
  {
    id: 'a1fc3d90-e825-11ed-b24d-893619994521',
    text: 'Help & Support',
  },
  {
    id: 'a4f929e0-e825-11ed-b24d-893619994521',
    text: 'iOS App',
  },
  {
    id: 'a7b0f6e0-e825-11ed-b24d-893619994521',
    text: 'Android App',
  },
]

export const socialLinks: SocialLinkProps[] = [
  {
    id: 'bb690ae0-8c1b-428c-9113-6e22cfb5c44b',
    text: 'Facebook',
    icon: 'facebook-f',
  },
  {
    id: '8083a63a-bd9c-4a89-b7ae-025e873529a8',
    text: 'Twitter',
    icon: 'twitter',
  },
  {
    id: '97bc93f4-ab8c-44dd-a33e-b8c5d6c8260b',
    text: 'TikTok',
    icon: 'tiktok',
  },
  {
    id: '42f20ef7-eabb-485c-8f69-4155f45623e4',
    text: 'Instagram',
    icon: 'instagram',
  },
  {
    id: '14c8210d-8242-48a5-ac7c-be9b5111b213',
    text: 'YouTube',
    icon: 'youtube',
  },
]
