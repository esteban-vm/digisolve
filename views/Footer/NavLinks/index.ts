export { default, navLinks } from './NavLinks'
