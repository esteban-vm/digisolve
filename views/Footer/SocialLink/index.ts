export { default /* type SocialLinkProps */ } from './SocialLink'
