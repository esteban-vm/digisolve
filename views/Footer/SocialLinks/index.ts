export { default, socialLinks } from './SocialLinks'
