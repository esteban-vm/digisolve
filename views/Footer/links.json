{
  "navLinks": [
    { "text": "About us" },
    { "text": "Blog" },
    { "text": "Help & Support" },
    { "text": "iOS App" },
    { "text": "Android App" }
  ],
  "socialLinks": [
    { "icon": ["fab", "facebook-f"], "title": "Facebook" },
    { "icon": ["fab", "twitter"], "title": "Twitter" },
    { "icon": ["fab", "google-plus-g"], "title": "Google Plus" },
    { "icon": ["fab", "instagram"], "title": "Instagram" },
    { "icon": ["fab", "youtube"], "title": "YouTube" }
  ]
}
