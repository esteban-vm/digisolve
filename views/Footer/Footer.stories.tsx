import type { Meta } from '@storybook/react'
import Footer from './Footer'

export default {
  component: Footer,
  title: 'Views/Footer',
} as Meta<typeof Footer>

export const FooterView = () => <Footer />
