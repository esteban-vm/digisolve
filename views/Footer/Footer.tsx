import type { Component } from '@/types'
import { Grid } from '@/components'
import { styled, mediaQuery } from '@/styles'
import { navLinks, socialLinks } from './Footer.data'
import NavLink from './NavLink'
import SocialLink from './SocialLink'

const FooterComponent: Component = (props) => {
  return (
    <footer {...props}>
      <Grid.Row>
        <Grid.Col isHalf>
          <ul className='nav__links'>
            {navLinks.map((link) => (
              <NavLink key={link.id} {...link} />
            ))}
          </ul>
        </Grid.Col>
        <Grid.Col isHalf>
          <ul className='social__links'>
            {socialLinks.map((link) => (
              <SocialLink key={link.id} {...link} />
            ))}
          </ul>
        </Grid.Col>
      </Grid.Row>
      <Grid.Row>
        <p>This webpage has been created for educational purposes.</p>
        <p>You are feel free to use it and customize it to make your own webpages.</p>
      </Grid.Row>
    </footer>
  )
}

const Footer = styled(FooterComponent)`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);

  ${mediaQuery('md')} {
    padding: 30px;
  }

  ul {
    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }
    }

    &[class$='__links'] {
      ${mediaQuery('xs')} {
        float: none;
        text-align: center;
        margin-bottom: 10px;
      }
    }
  }

  p {
    text-align: center;
    line-height: 1.5;

    ${mediaQuery('md')} {
      line-height: 1.3;
    }

    :first-of-type {
      margin-top: 50px;

      ${mediaQuery('md')} {
        margin-top: 20px;
      }
    }
  }

  .nav__links {
    float: left;
  }

  .social__links {
    float: right;
  }
`

export default Footer
