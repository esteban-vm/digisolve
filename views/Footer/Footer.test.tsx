import { render, cleanup, screen, create, axe } from '@/tests'
import { navLinks, socialLinks } from './Footer.data'
import { mediaQuery } from '@/styles'
import Footer from './Footer'

describe('ðŸ§ª FOOTER:', () => {
  describe('render tests:', () => {
    let container: HTMLElement

    beforeEach(() => {
      void ({ container } = render(<Footer />))
    })

    afterEach(cleanup)

    it('should be accessible', async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })

    it('all links should be rendered', () => {
      const linkElements = screen.getAllByRole('link')
      const numberOfLinks = navLinks.length + socialLinks.length
      expect(linkElements.length).toBe(numberOfLinks)

      for (let index = 0; index < numberOfLinks; index++) {
        const link = linkElements[index]
        expect(link).toBeInTheDocument()
        expect(link).toBeVisible()
      }
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Footer />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    describe('should have correct styles on:', () => {
      it('extra small screen devices', () => {
        const media = mediaQuery('xs')
        const target = "ul[class$='__links']"
        expect(tree).toHaveStyleRule('float', 'none', { target, media })
        expect(tree).toHaveStyleRule('text-align', 'center', { target, media })
        expect(tree).toHaveStyleRule('margin-bottom', '10px', { target, media })
      })

      it('medium screen devices', () => {
        const media = mediaQuery('md')
        expect(tree).toHaveStyleRule('padding', '30px', { media })
        expect(tree).toHaveStyleRule('line-height', '1.3', { target: 'p', media })
        expect(tree).toHaveStyleRule('margin-top', '20px', { target: 'p:first-of-type', media })
      })
    })
  })
})
