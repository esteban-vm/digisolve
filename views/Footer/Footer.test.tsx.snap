// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª FOOTER: style tests: should render with correct styles 1`] = `
.emotion-0 {
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);
}

@media only screen and (max-width: 1024px) {
  .emotion-0 {
    padding: 30px;
  }
}

.emotion-0 ul li {
  display: inline-block;
}

.emotion-0 ul li a {
  -webkit-transition: all 500ms;
  transition: all 500ms;
}

.emotion-0 ul li a:link,
.emotion-0 ul li a:visited {
  color: var(--color-dark);
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media only screen and (max-width: 480px) {
  .emotion-0 ul[class$='__links'] {
    float: none;
    text-align: center;
    margin-bottom: 10px;
  }
}

.emotion-0 p {
  text-align: center;
  line-height: 1.5;
}

@media only screen and (max-width: 1024px) {
  .emotion-0 p {
    line-height: 1.3;
  }
}

.emotion-0 p:first-of-type {
  margin-top: 50px;
}

@media only screen and (max-width: 1024px) {
  .emotion-0 p:first-of-type {
    margin-top: 20px;
  }
}

.emotion-0 .nav__links {
  float: left;
}

.emotion-0 .social__links {
  float: right;
}

.emotion-0 .social__links li:first-of-type a:hover {
  color: var(--color-facebook);
}

.emotion-0 .social__links li:nth-of-type(2) a:hover {
  color: var(--color-twitter);
}

.emotion-0 .social__links li:nth-of-type(3) a:hover {
  color: var(--color-tiktok);
}

.emotion-0 .social__links li:nth-of-type(4) a:hover {
  color: var(--color-instagram);
}

.emotion-0 .social__links li:last-of-type a:hover {
  color: var(--color-youtube);
}

.emotion-1 {
  margin-right: 20px;
}

.emotion-1 a:hover {
  opacity: 0.8;
}

.emotion-6 {
  margin-left: 20px;
}

.emotion-6 .icon {
  color: inherit;
}

<footer
  className="emotion-0"
>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_2 "
      role="presentation"
    >
      <ul
        className="nav__links"
      >
        <li
          className="emotion-1"
          id="97e51480-e825-11ed-b24d-893619994521"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            About us
          </a>
        </li>
        <li
          className="emotion-1"
          id="9c0412a0-e825-11ed-b24d-893619994521"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Blog
          </a>
        </li>
        <li
          className="emotion-1"
          id="a1fc3d90-e825-11ed-b24d-893619994521"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Help & Support
          </a>
        </li>
        <li
          className="emotion-1"
          id="a4f929e0-e825-11ed-b24d-893619994521"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            iOS App
          </a>
        </li>
        <li
          className="emotion-1"
          id="a7b0f6e0-e825-11ed-b24d-893619994521"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Android App
          </a>
        </li>
      </ul>
    </div>
    <div
      className="col span_1_of_2 "
      role="presentation"
    >
      <ul
        className="social__links"
      >
        <li
          className="emotion-6"
          id="bb690ae0-8c1b-428c-9113-6e22cfb5c44b"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Facebook"
          >
            <svg
              aria-hidden={true}
              className="icon"
              color="var(--color-primary)"
              fill="currentColor"
              height="1.5rem"
              role="img"
              viewBox="0 0 320 512"
              width="1.5rem"
            >
              <path
                d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              />
            </svg>
          </a>
        </li>
        <li
          className="emotion-6"
          id="8083a63a-bd9c-4a89-b7ae-025e873529a8"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Twitter"
          >
            <svg
              aria-hidden={true}
              className="icon"
              color="var(--color-primary)"
              fill="currentColor"
              height="1.5rem"
              role="img"
              viewBox="0 0 512 512"
              width="1.5rem"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </a>
        </li>
        <li
          className="emotion-6"
          id="97bc93f4-ab8c-44dd-a33e-b8c5d6c8260b"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="TikTok"
          >
            <svg
              aria-hidden={true}
              className="icon"
              color="var(--color-primary)"
              fill="currentColor"
              height="1.5rem"
              role="img"
              viewBox="0 0 448 512"
              width="1.5rem"
            >
              <path
                d="M448 209.91a210.06 210.06 0 01-122.77-39.25v178.72A162.55 162.55 0 11185 188.31v89.89a74.62 74.62 0 1052.23 71.18V0h88a121.18 121.18 0 001.86 22.17A122.18 122.18 0 00381 102.39a121.43 121.43 0 0067 20.14z"
              />
            </svg>
          </a>
        </li>
        <li
          className="emotion-6"
          id="42f20ef7-eabb-485c-8f69-4155f45623e4"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Instagram"
          >
            <svg
              aria-hidden={true}
              className="icon"
              color="var(--color-primary)"
              fill="currentColor"
              height="1.5rem"
              role="img"
              viewBox="0 0 448 512"
              width="1.5rem"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
        </li>
        <li
          className="emotion-6"
          id="14c8210d-8242-48a5-ac7c-be9b5111b213"
        >
          <a
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="YouTube"
          >
            <svg
              aria-hidden={true}
              className="icon"
              color="var(--color-primary)"
              fill="currentColor"
              height="1.5rem"
              role="img"
              viewBox="0 0 576 512"
              width="1.5rem"
            >
              <path
                d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <p>
      This webpage has been created for educational purposes.
    </p>
    <p>
      You are feel free to use it and customize it to make your own webpages.
    </p>
  </div>
</footer>
`;
